chapter.md

```diff
@@ -0,0 +1,735 @@
+  console.log(store.getState().color); // red, green, amber, red, ...
+}
+```
+
+If we want to add new capabilities,
+we simply update our reducer:
+
+```js
+function changeColor(state, action) {
+  switch(action.type) {
+
+    case 'NEXT':
+      ...as above...
+
+    case 'EMERGENCY':
+      return {color: 'red'};
+
+    default:
+      return state;
+  }
+}
+```
+
+This may seem like a lot of work to manage a single traffic light,
+but that work pays off as soon as we have to worry about asynchronous updates.
+For example,
+if we want to do an emergency test of the light every night at 1:00 am,
+all we have to do is this:
+
+```js
+let delay = until(tomorrow() + ONE_HOUR);
+setTimeout(() => changeColor(state, {type: 'EMERGENCY'}), delay);
+```
+
+When the timeout callback is triggered,
+Redux will put the traffic light in the required state
+regardless of what else has gone on or is going on.
+
+## Setting Up Forms to Work with Redux
+
+As our running example,
+we will build a set of forms that allow users to create a wizard for a role-playing game.
+This example was built by Renee Vrantsidis and Daniel Figueiredo
+for [the talk][repo-talk] this chapter is based on,
+and the complete application is [available on GitHub][repo-application].
+
+The first step is to plan the shape of our forms.
+We can structure the store however we want,
+but since we know that real applications evolve,
+we will define things using TypeScript interfaces
+rather than relying on specific concrete classes,
+since that will make it easier to swap things in and out for testing later on.
+
+```ts
+export interface IForm {
+  character: ICharacter;        // our single top-level object (for now)
+}
+
+export interface ICharacter {
+  name?: string;                // optional name
+  bioSummary: IBioSummary;      // biographical information (see below)
+  skills: string[];             // list of skills
+}
+
+export interface IBioSummary {
+  age: number;                  // age in years
+  alignment: string;            // good or bad, lawful or chaotic
+  race: string;                 // character's species
+}
+```
+
+These interfaces define a store
+that contains a single object representing a character,
+rather than using the character itself as the store.
+That way,
+if we want to add other top-level items in future,
+we won't need to reorganize existing material.
+For example,
+the next version of our application might have this structure:
+
+```ts
+export interface IForm {
+  character: ICharacter;        // character info
+  equipment: IEquipment;        // character's equipment
+}
+```
+
+One consequence of this decision is that
+every action should contain the path to the particular part of the state it applies to.
+Agile purists might say that we shouldn't add this until we need it,
+but we have enough experience with applications of this kind to know that we're going to,
+and adding that to our architecture from the start will save us re-work later.
```

Comment:
That said, having an application with just a login form would certainly make this an overhead 
^ our common actions and reducers tries to solve enterprises and large scale form apps =]